src/main/java/io/cdap/plugin/snowflake/common/client/SnowflakeAccessor.java

```diff
@@ -111,6 +111,9 @@ private void initDataSource(SnowflakeBasicDataSource dataSource, BaseSnowflakeCo
     if (config.getOauth2Enabled()) {
       String accessToken = OAuthUtil.getAccessTokenByRefreshToken(HttpClients.createDefault(), config);
       dataSource.setOauthToken(accessToken);
+      // Added to fix the issue on the driver side code. This is mandatory field just for validation,
+      // value of password is not getting checked
+      dataSource.setPassword(accessToken);
```

Comment:
Please add a comment in place about the details about why we need to call setPassword.

Reply:
Added. Also added the link for this issue discussion on snowflake repo.